File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:

        def getArea(p1, p2) -> int:
            h = p1[1] if p1[1] < p2[1] else p2[1]
            l = abs(p1[0] - p2[0])
            return h * l
            
        l = 0
        r = len(height) -1
        answer = 0

        while l < r:
            answer = max(getArea([l, height[l]], [r, height[r]]), answer)

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
                    
        return answer

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int r = height.length-1;
        int l = 0;
        int answer = 0;
        while (l < r) {
            int container = (r - l) * Math.min(height[r], height[l]);

            answer = Math.max(answer, container);

            if (height[r] < height[l]){
                r--;
            } else {
                l++;
            }
        }

        return answer;
    }
}